fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/TemplateDruidQuery.java

```diff
@@ -381,6 +387,50 @@ public TemplateDruidQuery merge(TemplateDruidQuery sibling) {
         return new LinkedHashSet<>(resultAggregationsByName.values());
     }
 
+    /**
+     * Given two sets of Virtual Columns, merge them into a single set, combining where possible.
+     *
+     * @param set1  First set of Virtual Columns
+     * @param set2  Second set of VirtualColumns
+     *
+     * @return the merged Virtual Columns
+     */
+    private LinkedHashSet<VirtualColumn> mergeVirtualColumns(
```

Comment:
Needs a test on this method to verify 
1- all distinct ones are maintained.
2- duplicates that have the same meaning generate errors
3- duplicates with the same name do not generate errors.